lib/extract-stream.js

```diff
@@ -9,28 +9,33 @@ function extractStream (dest, opts) {
   const sawIgnores = new Set()
   return tar.x({
     cwd: dest,
-    filter: (name, entry) => !entry.header.type.match(/^.*link$/i),
+    // Use endsWith rather than match with regex. Should be faster.
+    filter: (name, entry) => !entry.header.type.endsWith('link'),
     strip: 1,
     onwarn: msg => opts.log && opts.log.warn('tar', msg),
     uid: opts.uid,
     gid: opts.gid,
     onentry (entry) {
-      if (entry.type.toLowerCase() === 'file') {
+      // Only lowerCase once.
+      const entryType = entry.type.toLowerCase()
+      if (entryType === 'file') {
         entry.mode = opts.fmode & ~(opts.umask || 0)
-      } else if (entry.type.toLowerCase() === 'directory') {
+      } else if (entryType === 'directory') {
```

Comment:
the code still lowercases it, on [line 20](https://github.com/zkat/pacote/pull/114/files/bdfd3b4fa84c1c4a7abb65d79e2eec4cb104ebbe#diff-4b98eff17b1a23d958e12e0fa06015a8R20), but only once for the two (potentially three) times it's compared rather than every time.